src/workerd/io/worker-interface.h

```diff
@@ -38,6 +38,11 @@ class WorkerInterface: public kj::HttpService {
   //   more resources to be dropped when merely proxying a request. However, it means we would no
   //   longer be immplementing kj::HttpService. But maybe that doesn't matter too much in practice.
 
+  virtual kj::Promise<void> connect(kj::StringPtr host,
+                                    const kj::HttpHeaders& headers,
+                                    kj::AsyncIoStream& connection,
+                                    ConnectResponse& response) = 0;
```

Comment:
Maybe leave a comment:

> This is the same as the inherited `HttpService::connect()`, but we override it to be pure-virtual to force all subclasses of `WorkerInterface` to implement it explicitly rather than get the default implementation which throws an unimplemented exception.